discovery/src/main/java/io/airlift/discovery/client/DiscoveryBinder.java

```diff
@@ -104,15 +117,90 @@ public void bindHttpSelector(ServiceType serviceType)
         binder.bind(HttpServiceSelector.class).annotatedWith(serviceType).toProvider(new HttpServiceSelectorProvider(serviceType.value())).in(Scopes.SINGLETON);
     }
 
+    public static class HttpServiceAnnouncementBuilder
```

Comment:
End the name with `BindingBuilder` to match the convention. We could shorten it to `HttpAnnoucementBindingBuilder`, matching the name of `HttpAnnouncementProvider`, with the reasoning that `service` is generic and is replaced with `http`.